tests/Backend/CommonPart1/TableExporterTest.php

```diff
@@ -84,6 +85,56 @@ public function testLimitParameter()
         $this->assertCount($exportOptions['limit'], $parsed);
     }
 
+
+    public function testExportTables()
+    {
+        $filesBasePath = __DIR__ . '/../../_data/';
+        $table1Id = $this->_client->createTableAsync($this->getTestBucketId(self::STAGE_IN), 'languages1', new CsvFile($filesBasePath . 'languages.csv'));
+        $table2Id = $this->_client->createTableAsync($this->getTestBucketId(self::STAGE_IN), 'languages2', new CsvFile($filesBasePath . 'languages.csv'));
+        $exporter = new TableExporter($this->_client);
+        $file1 = __DIR__ . '/../../_tmp/languages1.csv';
+        $file2 = __DIR__ . '/../../_tmp/languages2.csv';
+        $exports = array(
+            array(
+                'tableId' => $table1Id,
+                'destination' => $file1
+            ),
+            array(
+                'tableId' => $table2Id,
+                'destination' => $file2
+            )
+
+        );
+        $exporter->exportTables($exports);
+        // compare data
+        $this->assertTrue(file_exists($file1));
+        $this->assertTrue(file_exists($file2));
+        $this->assertLinesEqualsSorted(file_get_contents($filesBasePath . 'languages.csv'), file_get_contents($file1), 'imported data comparison');
+        $this->assertLinesEqualsSorted(file_get_contents($filesBasePath . 'languages.csv'), file_get_contents($file2), 'imported data comparison');
+    }
+
+    public function testExportTablesMissingTableId()
+    {
+        $exporter = new TableExporter($this->_client);
+        try {
+            $exporter->exportTables(array(array()));
+            $this->fail('Missing exception');
+        } catch (Exception $e) {
+            $this->assertEquals('Missing tableId', $e->getMessage());
+        }
+    }
+
+    public function testExportTablesMissingDestination()
+    {
+        $exporter = new TableExporter($this->_client);
+        try {
+            $exporter->exportTables(array(array('tableId' => 'dummy')));
+            $this->fail('Missing exception');
+        } catch (Exception $e) {
+            $this->assertEquals('Missing destination', $e->getMessage());
+        }
+    }
+
```

Comment:
Chybí mi test na to, jak se to asynchronní volání chová, pokud to vyhodí chybu na straně KBC. Tj. když tam vyplníš všechno potřebné, ale nesmysly.

Reply:
Ideálně třeba platná tabulka a pak nějaký bug... aby bylo vidět, co se stane...

Reply:
to by všechno měly pokrýt existující testy `\Keboola\StorageApi\Client::exportTableAsync`, využívají totožný vnitřnosti

Reply:
přihodil jsem test na neexistující tabulku